src/main/webapp/js/science_portal_core.js

```diff
@@ -0,0 +1,325 @@
+;(function($) {
+  // register namespace
+  $.extend(true, window, {
+    cadc: {
+      web: {
+        science: {
+          portal: {
+            core: {
+              PortalCore: PortalCore,
+              // Events
+              events: {
+                onAuthenticated: new jQuery.Event('sciPort:onAuthenticated'),
+                onServiceURLOK: new jQuery.Event('sciPort:onServiceURLOK'),
+                onServiceURLFail: new jQuery.Event('sciPort:onServiceURLFail')
+              }
+            }
+          }
+        }
+      }
+    }
+  })
+
+  /**
+   * Common functions to support Science Portal UI.
+   *
+   * @constructor
+   * @param {{}} inputs   Input configuration.
+   * @param {String} [inputs.baseURL='https://www.canfar.net/'] URL of the /reg web service
+   * needed by the Registry to look up web service and ui URLs for use in ajax calls by this page.
+   */
+  function PortalCore(inputs) {
+
+    var _selfPortalCore = this
+    var baseURL =
+            inputs && inputs.hasOwnProperty('baseURL')
+                ? inputs.baseURL
+                : 'https://www.canfar.net'
+
+    var _registryClient = new Registry({
+      baseURL: baseURL
+    })
+
+    var _ajaxCallCount = 0
+
+    var _sessionServiceResourceID = inputs.sessionsResourceID
+
+    // Initialize the context help, kick off
+    // loading the content file, and load the popovers
+    var _chOptions = {
+      contentFile: 'contexthelp_en.json',
+      clickClassname: 'science-portal-tooltip',
+      hoverClassname: 'science-portal-tooltip-btn'
+    }
+    var _contextHelp = new cadc.web.CadcContextHelp(_chOptions)
+    _contextHelp.init()
+
+    function init() {
+      setSessionServiceURL()
+    }
+
+    // ------------ Page state management functions ------------
+
+    function clearAjaxAlert() {
+      $('.alert-danger').addClass('hidden')
+      $('.alert-success').addClass('hidden')
+      setProgressBar('okay')
+    }
+
+    // Communicate AJAX progress and status using progress bar
+    function setProgressBar(state) {
+      var _progressBar = $('.sp-progress-bar')
+      switch (state) {
+        case 'busy': {
+          _progressBar.addClass('progress-bar-striped')
+          _progressBar.removeClass('progress-bar-danger')
+          _progressBar.addClass('progress-bar-success')
+          break
+        }
+        case 'okay': {
+          _progressBar.removeClass('progress-bar-striped')
+          _progressBar.removeClass('progress-bar-danger')
+          _progressBar.addClass('progress-bar-success')
+          break
+        }
+        case 'error': {
+          _progressBar.removeClass('progress-bar-striped')
+          _progressBar.removeClass('progress-bar-success')
+          _progressBar.addClass('progress-bar-danger')
+          break
+        }
+        default: {
+          // Nothing
+          break
+        }
+      }
+    }
```

Comment:
This progress bar looks familiar.  If we're going to use this progress bar in multiple places, we may want to extract it into something shareable.

Reply:
It's originally from User Storage. I could put it into the CANFAR ROOT.war (wwebui/canfar-root) project, but that increases dependence on an internal git library. Do we have an actual open source library to put this kind of function in? If not, I'll put it alongside the code for handling the context help (up in canfar-root.)

Reply:
Created cadc.progressbar.js in cadc-9210 branch of wwebui/canfar-root that needs to be reviewed as well.